dbms/src/Encryption/RateLimiter.cpp

```diff
@@ -624,17 +596,28 @@ void IORateLimiter::setStop()
 
 void IORateLimiter::runAutoTune()
 {
-    auto auto_tune_worker = [&]() {
+    auto auto_tune_and_get_io_info_worker = [&]() {
+        using time_point = std::chrono::time_point<std::chrono::system_clock>;
+        using clock = std::chrono::system_clock;
+        time_point auto_tune_time = clock::now();
+        time_point update_io_stat_time = auto_tune_time;
         while (!stop.load(std::memory_order_relaxed))
         {
-            ::sleep(io_config.auto_tune_sec > 0 ? io_config.auto_tune_sec : 1);
-            if (io_config.auto_tune_sec > 0)
+            std::this_thread::sleep_for(std::chrono::milliseconds(update_io_stat_period_ms));
+            auto now_time_point = clock::now();
+            if ((io_config.auto_tune_sec > 0) && (now_time_point - auto_tune_time > std::chrono::seconds(io_config.auto_tune_sec)))
             {
                 autoTune();
+                auto_tune_time = now_time_point;
+            }
+            if ((bg_read_limiter || fg_read_limiter) && (now_time_point - update_io_stat_time > std::chrono::milliseconds(update_io_stat_period_ms)))
```

Comment:
This thread sleep for `update_io_stat_period_ms` milliseconds and wake up. `now_time_point - update_io_stat_time` is very likely equal to `update_io_stat_period_ms`.

Reply:
get, I will change `(now_time_point - update_io_stat_time > std::chrono::milliseconds(update_io_stat_period_ms))` to  `likely(now_time_point - update_io_stat_time >= std::chrono::milliseconds(update_io_stat_period_ms))`

Reply:
What about just sleep `update_io_stat_period_ms` milliseconds and not to check `now_time_point - update_io_stat_time` here or check `now_time_point - update_io_stat_time >= update_io_stat_period_ms`?